test/unit/Elsa.Activities.UnitTests/Branching/IfTests.cs

```diff
@@ -0,0 +1,107 @@
+using Elsa.Extensions;
+using Elsa.Testing.Shared;
+using Elsa.Workflows;
+
+namespace Elsa.Activities.UnitTests.Branching;
+
+public class IfTests
+{
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task Should_Set_Result_To_Condition_Value_Regardless_Of_Branch_Presence(bool conditionValue)
+    {
+        // Arrange - Test with no branches to verify result is independent of branch activities
+        var ifActivity = new If(() => conditionValue);
+
+        // Act
+        var context = await ExecuteAsync(ifActivity);
+
+        // Assert
+        var resultValue = (bool)context.GetActivityOutput(() => ifActivity.Result)!;
+        Assert.Equal(conditionValue, resultValue);
+    }
+
+    [Theory]
+    [InlineData(true, true, false)] // condition true, has then branch, no else branch
+    [InlineData(false, false, true)] // condition false, no then branch, has else branch
+    public async Task Should_Set_Result_Correctly_With_Branch_Configuration(bool conditionValue, bool hasThenBranch, bool hasElseBranch)
+    {
+        // Arrange
+        var ifActivity = new If(() => conditionValue);
+        
+        if (hasThenBranch)
+        {
+            // Using a simple WriteLine activity to avoid variable complexity
+            ifActivity.Then = new WriteLine(new Input<string>("then executed"));
+        }
+        
+        if (hasElseBranch)
+        {
+            ifActivity.Else = new WriteLine(new Input<string>("else executed"));
+        }
```

Comment:
Although this test assigns the Then or Else branch conditionally, the assertion is unrelated to this and so this setup seems to bear no meaning. Effectively, this test is the same one as `Should_Set_Result_To_Condition_Value_Regardless_Of_Branch_Presence`